Observer/OfflinePaymentObserver.php

```diff
@@ -76,7 +76,14 @@ public function execute(Observer $observer)
                 $barcodeHtml   = "<img src= '".$charge->source['scannable_code']['image']['download_uri']."'/>";
                 $emailTemplate = 'send_email_paynow_template';
                 $emailData->setData(['banksUrl' => $this->_assetRepo->getUrl('Omise_Payment::images/paynow_supportedbanks.png')]);
-                $validUntil    = date("d-m-Y H:i:s" , strtotime($charge->expires_at));
+                $validUntil = $this->getPaynowChargeExpiryTime();
+                break;
+            case 'promptpay':
+                // make sure timezone is Thailand.
+                date_default_timezone_set("Asia/Bangkok");
+                $barcodeHtml   = "<img src= '".$charge->source['scannable_code']['image']['download_uri']."'/>";
+                $emailTemplate = 'send_email_promptpay_template';
+                $validUntil    = date("d-m-Y H:i:s" , strtotime('+1 day'));
```

Comment:
In this kind of code, wouldn't it be better to have an email-service class separately instead of leaving this inside the Observer class?

Reply:
Will do this in future refactoring PRs.

Reply:
Just wonder if we agreed that this will eventually get refactoring in the future, then would it be better if we just do it right away?